autonomy/analyse/service.py

```diff
@@ -375,6 +455,117 @@ def cross_verify_overrides(
             check_with_name=str(skill_config.package_id),
         )
 
+    @classmethod
+    def validate_override_env_vars(
+        cls,
+        overrides: Union[OrderedDict, dict],
+        validate_env_var_name: bool = False,
+        json_path: Optional[List[str]] = None,
+    ) -> List[str]:
+        """Validate environment variables."""
```

Comment:
This check will make sure that

- All overrides are defined as the env vars at agent layer
- All overrides are defined as the env vars with env var name defined at the service layer
- Env vars can be loaded, data types and default values are defined properly

and prevent issues like this https://github.com/valory-xyz/open-autonomy/issues/1898#issuecomment-1493139762 and https://github.com/valory-xyz/open-autonomy/issues/1898#issuecomment-1497256903